src/displayapp/widgets/Counter.cpp

```diff
@@ -0,0 +1,105 @@
+#include "displayapp/widgets/Counter.h"
+
+using namespace Pinetime::Applications::Widgets;
+
+Counter::Counter(int min, int max) : min {min}, max {max} {
+}
+
+void Counter::upBtnEventHandler(lv_obj_t* obj, lv_event_t event) {
+  auto* widget = static_cast<Counter*>(obj->user_data);
+  if (event == LV_EVENT_SHORT_CLICKED || event == LV_EVENT_LONG_PRESSED_REPEAT) {
+    widget->Increment();
+  }
+}
+
+void Counter::downBtnEventHandler(lv_obj_t* obj, lv_event_t event) {
+  auto* widget = static_cast<Counter*>(obj->user_data);
+  if (event == LV_EVENT_SHORT_CLICKED || event == LV_EVENT_LONG_PRESSED_REPEAT) {
+    widget->Decrement();
+  }
+}
+
+void Counter::Increment() {
+  value++;
+  if (value > max) {
+    value = min;
+  }
+  UpdateLabel();
+};
+
+void Counter::Decrement() {
+  value--;
+  if (value < min) {
+    value = max;
+  }
+  UpdateLabel();
+};
+
+void Counter::SetValue(int newValue) {
+  value = newValue;
+  UpdateLabel();
+}
+
+void Counter::UpdateLabel() {
+  lv_label_set_text_fmt(number, "%.2i", value);
+}
+
+void Counter::Create() {
+  counterContainer = lv_obj_create(lv_scr_act(), nullptr);
+  lv_obj_set_style_local_bg_color(counterContainer, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_MAKE(0x38, 0x38, 0x38));
+
+  number = lv_label_create(counterContainer, nullptr);
+  lv_obj_set_style_local_text_font(number, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_76);
+  lv_obj_align(number, nullptr, LV_ALIGN_CENTER, 0, 0);
+  lv_obj_set_auto_realign(number, true);
+  lv_label_set_text_static(number, "00");
+
+  static const uint8_t padding = 5;
```

Comment:
we could make this variable constexpr

```suggestion
  constexpr uint8_t padding = 5;
```

Reply:
Would this be preferrable? I don't think there's a difference in the result.

Reply:
You're right, the end result with optimization will most likely be the same

for the `static const` the compiler will create a global variable, which with high probability will be optimized out later

with `constexpr` the variable will not be created in the first place (as far as I understand). It's more explicit to write `constexpr` to express what we actually want instead of relying on the knowledge that the optimizer will remove the variable in the end

Reply:
I changed it to static constexpr to force compile time initialization, since just constexpr is technically just a suggestion.